Use all 4 vCPUs on GitHub-hosted runners

tautschnig opened the pull request:
Since January GitHub-hosted runners feature a higher vCPU count for Linux and Windows, and also for macos-13. See
https://github.blog/2024-01-17-github-hosted-runners-double-the-power-for-open-source/ for the announcement.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Use all 4 vCPUs on GitHub-hosted runners

Since January GitHub-hosted runners feature a higher vCPU count for
Linux and Windows, and also for macos-13. See
https://github.blog/2024-01-17-github-hosted-runners-double-the-power-for-open-source/
for the announcement.

TGWDB reviewed (approved):
I'll approve, but considering all the changes this looks like a refactor to parameterise this value and declare it once might be worth it.

tautschnig commented:
> I'll approve, but considering all the changes this looks like a refactor to parameterise this value and declare it once might be worth it.

Thank you, you're absolutely right, I'll add this to this PR.

tautschnig committed:
GitHub CI: Move number of Linux/Windows vCPUs to config variable

The churn in the preceding commit should be avoided for future
infrastructure upgrades.

tautschnig committed:
Update job execution times

Document the speed-up afforded by additional vCPUs
